### PR TITLE
_dump_date zfills years < 4 digits

### LINKED ISSUE #1739: _dump_date year conversion doesn't zfill(4)
I have some dates that have a year before year 1000 and the _dump_date function doesn't represent them as zero filled to 4 so parsing the output with '%a, %d %b %Y %H:%M:%S %Z' throws an exception.

```
index b428ceeb..d233d0f2 100644
--- a/src/werkzeug/http.py
+++ b/src/werkzeug/http.py
@@ -882,7 +882,7 @@ def _dump_date(d, delim):
             "Dec",
         )[d.tm_mon - 1],
         delim,
-        str(d.tm_year),
+        str(d.tm_year).zfill(4),
         d.tm_hour,
         d.tm_min,
         d.tm_sec,
```

### PR DESCRIPTION
**Pull request by nicolary**
fixes #1739

Added a zfill to correctly display year on dates before year 1000.

### CODE REVIEW

**Comment by nicolary** [comment]
Sorry I screwed up the base branch...

File: tests/test_http.py [hunk 1]
```diff
@@ -668,6 +669,39 @@ def test_content_range_parsing(self):
         assert rv.length == 100
         assert rv.units == "bytes"
 
+    def test_http_date_lt_1000(self):
```
Comment (davidism) [inline tests/test_http.py]: This should use `pytest.mark.parametrize`.

**Comment by davidism** [comment]
Can you also add a changelog to the 1.0.1 section?

**Comment by nicolary** [comment]
Hi - my tox run failed on "Job Python 3.8 Mac" with:
```
2020-02-26T00:35:31.7220500Z =================================== FAILURES ===================================
2020-02-26T00:35:31.7223060Z _______________________________ test_basic_local _______________________________
2020-02-26T00:35:31.7224700Z tests/test_local.py:36: in test_basic_local
2020-02-26T00:35:31.7226140Z     assert sorted(values) == [1, 2, 3]
2020-02-26T00:35:31.7226850Z E   assert [1, 3] == [1, 2, 3]
2020-02-26T00:35:31.7227430Z E     At index 1 diff: 3 != 2
2020-02-26T00:35:31.7229070Z E     Right contains one more item: 3
2020-02-26T00:35:31.7230290Z E     Use -v to get the full diff
```

This has nothing to do with my code additions - was it just a hiccup and the tests can be re-run?

**Comment by davidism** [comment]
Yeah, those are just flaky tests, don't worry about that.

**Comment by nicolary** [comment]
Ok - let me know if I need to do anything else. Thanks for the help.

**Comment by nicolary** [comment]
thank you for your service

**Pull request merged**